client.go

```diff
@@ -479,84 +451,37 @@ func (c *Client) Subscribe(params SubscriptionParameters) (*Subscription, error)
 		res.RevisedPublishingInterval,
 		res.RevisedLifetimeCount,
 		res.RevisedMaxKeepAliveCount,
-		make(chan PublishNotificationData, params.ChannelBufferSize),
-		c.PublishLoop(),
+		params.Notifs,
+		c,
 	}
 	c.subscriptions[sub.SubscriptionID] = sub
 
 	return &sub, nil
 }
 
-func (c *Client) CreateSubscription(req *ua.CreateSubscriptionRequest) (*ua.CreateSubscriptionResponse, error) {
-	var res *ua.CreateSubscriptionResponse
-	err := c.Send(req, func(v interface{}) error {
-		return safeAssign(v, &res)
-	})
-	return res, err
-}
-
-// Unsubscribe() deletes the given Subscription from server and stops the Publish loop that was
-// started by Subscribe()
-func (c *Client) Unsubscribe(sub *Subscription) error {
-	if registeredSub, ok := c.subscriptions[sub.SubscriptionID]; ok {
-		close(registeredSub.stopPublishLoop)
-		delete(c.subscriptions, sub.SubscriptionID)
+func setDefaultSubscriptionParameters(params *SubscriptionParameters) {
+	if params.MaxNotificationsPerPublish == 0 {
+		params.MaxNotificationsPerPublish = 10000
 	}
-
-	res, err := c.DeleteSubscriptions([]uint32{sub.SubscriptionID})
-	if err != nil {
-		return err
+	if params.LifetimeCount == 0 {
+		params.LifetimeCount = 10000
 	}
-	if res.ResponseHeader.ServiceResult != ua.StatusOK {
-		return res.ResponseHeader.ServiceResult
+	if params.MaxKeepAliveCount == 0 {
+		params.MaxKeepAliveCount = 3000
 	}
-
-	return nil
-}
-
-func (c *Client) DeleteSubscriptions(subIds []uint32) (*ua.DeleteSubscriptionsResponse, error) {
-	req := &ua.DeleteSubscriptionsRequest{
-		SubscriptionIDs: subIds,
+	if params.Interval == 0 {
+		params.Interval = 100 * time.Millisecond
 	}
-	var res *ua.DeleteSubscriptionsResponse
-	err := c.Send(req, func(v interface{}) error {
-		return safeAssign(v, &res)
-	})
-	return res, err
-}
-
-func NewMonitoredItemCreateRequestWithDefaults(nodeID *ua.NodeID, attributeID ua.AttributeID, clientHandle uint32) *ua.MonitoredItemCreateRequest {
-	if attributeID == 0 {
-		attributeID = ua.AttributeIDValue
-	}
-	readValueID := &ua.ReadValueID{
-		NodeID:       nodeID,
-		AttributeID:  attributeID,
-		DataEncoding: &ua.QualifiedName{},
-	}
-	params := ua.MonitoringParameters{
-		ClientHandle:     clientHandle,
-		DiscardOldest:    true,
-		Filter:           nil,
-		QueueSize:        10,
-		SamplingInterval: 0.0,
+	if params.Notifs == nil {
+		params.Notifs = make(chan PublishNotificationData)
```

Comment:
I don't have a preference either way. What's your reasoning?

Reply:
Generally, I always have to think about side effects (or lack thereof) when using value instead of pointer semantics. For one, (I think) all calls in the API use struct pointers and this one is different. So there should be a reason for this. Otherwise, it just trips up users of the API. 

Two, the struct is going to be copied when passed around but the embedded value is an `interface{}`. So you never create a deep copy. 

I tend to use value semantics for structs when I don't want users to change the values or trip them up if they do, like for a `Config` struct. For the rest I tend to go with pointers unless there is a good reason not to.

If you don't have a preference then I would aim for consistency with the rest of the API, hence pointer.

Reply:
Points taken. I'll change to `*PublishNotificationData`